python/pfs/drp/stella/fitPfsFluxReference.py

```diff
@@ -0,0 +1,978 @@
+            # Convert selected tics to indices in paramCatalog
+            indices = np.array([
+                paramToIndex[tuple(ticToParam[i][t] for i, t in enumerate(tic))]
+                for tic in ticList
+            ], dtype=int)
+            # Cut only necessary portions out of pdf and paramCatalog.
+            cutProb = pdf[indices]
+            cutParamCatalog = paramCatalog[indices]
+
+            # Fit y = \sum_{i+j <= 2} coeff[i,j] x[i] x[j];
+            # where y = pdf, x[0] = 1, x[1:] = (parameters)
+            axisList = [1.0] + [cutParamCatalog[name] for name in paramNames]
+            axisToIndex = {}
+            M = np.empty(shape=(len(cutProb), len(axisList)*(len(axisList) + 1)//2), dtype=float)
+            k = 0
+            for i in range(0, len(axisList)):
+                for j in range(i, len(axisList)):
+                    M[:, k] = axisList[i] * axisList[j]
+                    axisToIndex[i, j] = k
+                    k += 1
+
+            coeff = np.linalg.lstsq(M, cutProb, rcond=None)[0]
+
+            # Reorder the terms of the polynomial
+            # from y = \sum_{i+j <= 2} coeff[i,j] x[i] x[j]
+            # to y = x'^T A x' + b^T x' + c
+            # where x' = x[1:]
+            axisList = axisList[1:]
+            A = np.empty(shape=(len(axisList), len(axisList)), dtype=float)
+            for i in range(0, len(axisList)):
+                A[i, i] = coeff[axisToIndex[i + 1, i + 1]]
+                for j in range(i + 1, len(axisList)):
+                    A[i, j] = A[j, i] = 0.5 * coeff[axisToIndex[i + 1, j + 1]]
+
+            b = np.empty(shape=(len(axisList)), dtype=float)
+            for i in range(0, len(axisList)):
+                b[i] = coeff[axisToIndex[0, i + 1]]
+
+            # Now we know the peak is at -(1/2) A^{-1} b,
+            # but we have to be careful whether it is a valid solution.
+            bestParam = None
+            if np.linalg.det(A) > 0:
+                peak = -0.5 * np.linalg.solve(A, b)
+                # We employ this peak only if it is within the minimum box
+                # that includes ticList.
+                if all(
+                        ticToParam[i][min(tic[i] for tic in ticList)] <= p <=
+                        ticToParam[i][max(tic[i] for tic in ticList)]
+                        for i, p in enumerate(peak)):
+                    bestParam = tuple(peak) + tuple(peakParam[name] for name in fixedParamNames)
+
+            if bestParam is None:
+                bestParam = tuple(peakParam[name] for name in paramNames + fixedParamNames)
+
+            wavelength, flux = self.modelInterpolator.interpolate(*bestParam)
+
+            target = Target(0, 0, "0,0", 0)
+            mask = np.zeros(shape=wavelength.shape, dtype=int)
+            flags = MaskHelper()
+            mask[:] = np.where(np.isfinite(flux), 0, flags.add("BAD"))
+            spectrum = PfsSimpleSpectrum(target, wavelength, flux, mask, flags)
```

Comment:
I thought this was supposed to be a `PfsFluxReference` or something like that? I'm concerned that the need to lie to the datamodel (a fake `Target`) indicates that there's something missing from the datamodel.

Reply:
I'm trying to replace `List[PfsSimpleSpectrum]` with `PfsFluxReference`. To be frank, I think there should be a simpler `PfsSimpleSpectrum` and `PfsFluxArray(Set)` that do not have target or identity, and the current `PfsSimpleSpectrum` and `PfsFluxArray(Set)` should be subclasses of the simpler ones. There should also be a simple (no target, no identity) `PfsSimpleSpectrumSet`, of which `PfsFluxReference` should be a subclass.

Reply:
That makes sense. I agree.